pkg/util/check.go

```diff
@@ -30,10 +30,14 @@ import (
 var minPDVersion *semver.Version = semver.New("4.0.0-rc.1")
 var minTiKVVersion *semver.Version = semver.New("4.0.0-rc.1")
 
-func removeV(v string) string {
+func removeVAndHash(v string) string {
 	if v == "" {
 		return v
 	}
+	hash := regexp.MustCompile("-[0-9]+-g[0-9a-f]{8,}")
```

Comment:
After this, comparing between `4.0.0-8-g30f0b014` and `4.0.0-9-g31dae220` seems not work

Reply:
They are the same. Semver should not include `-8-g30f0b014`.

Reply:
According to [semver.org](https://semver.org/#spec-item-9), `4.0.0-8-g30f0b014` is a pre-release version. Well as we don't need to compare `4.0.0-8-g30f0b014` and `4.0.0-9-g31dae220`, it looks all right to treat `4.0.0-8-g30f0b014` as `4.0.0`
(maybe `4.0.0+8-g30f0b014` is better, where `+8-g30f0b014` is used as `build metadata`)